aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -1008,5 +1024,29 @@ Tensor masked_softmax_cuda(const Tensor& input, const Tensor& mask) {
     }
     return output;
 }
+
+Tensor masked_softmax_backward_cuda(
+    const Tensor& grad,
+    const Tensor& output,
+    int64_t dim,
+    const Tensor& mask) {
```

Comment:
only some of the paths in host_softmax_backward are patched to handle mask, yet you don't check here that those paths will be taken, and don't provide fallback.

Reply:
This should be fixed now!